src/rgw/setup/rgw.py

```diff
@@ -322,15 +334,36 @@ def _check_consul_connection(conf: MappedConf):
             raise SetupError(errno.EINVAL, f"Consul server {host:port} not reachable")
 
     @staticmethod
-    def _fetch_endpoint_url(conf: MappedConf, confstore_endpoint_key: str, endpoint_type: str):
+    def _get_gconf_key_list(conf: MappedConf, gconf_num_key:str, actual_gconf_key:str):
+        """Get value list of specified gconf key."""
+        Log.info(f'Fetching gconf num_key from Gconf: {gconf_num_key}')
+        num_of_keys = int(Rgw._get_cortx_conf(conf, gconf_num_key))
+        Log.info(f'Found number of keys:{num_of_keys}')
+        if num_of_keys == 0:
+            raise SetupError(errno.EINVAL, f"Invalid/Missing values found in gconf for key :'{gconf_num_key}'")
+        value_list = []
+        for value_index in range(0, num_of_keys):
+            key_value = Rgw._get_cortx_conf(conf, actual_gconf_key % value_index)
+            value_list.append(key_value)
+        Log.info(f'Returing value list: {value_list}')
+        return value_list
+
+    @staticmethod
+    def _fetch_consul_endpoint_url(conf: MappedConf, endpoint_type: str):
         """Fetch endpoint url based on endpoint type from cortx config."""
-        endpoints = Rgw._get_cortx_conf(conf, confstore_endpoint_key)
-        endpoints_values = list(filter(lambda x: urlparse(x).scheme == endpoint_type, endpoints))
-        if len(endpoints_values) == 0:
+        Log.info('Fetching consul endpoint url using new gconf method.')
+        consul_endpoints = Rgw._get_gconf_key_list(conf, const.CONSUL_NUM_ENDPOINT_KEY,
```

Comment:
should be under try catch block, as it raises exception @ line 343

Reply:
@cdeshmukh If we see any issue while fetching the key-values from Gconf then it should terminate the deployment with error message hence we are just raising SetupError at 343 line and push this SetupError to caller without handling it try-catch block.

Reply:
Ok, consul is must now for RGW from this PR, given GConf will reside there. If yes, i understand

Reply:
In above code context, we are using consul endpoint to save rgw lock variable.hence we are fetching the consul endpoint.
its not related to this Gconf migration to consul.